desc/optimize/optimizer.py

```diff
@@ -494,7 +498,7 @@ def jac(x_reduced):
                 x0=x0_reduced,
                 grad=grad_wrapped,
                 args=(),
-                method=method,
+                method=self.method,
```

Comment:
i think this needs to be method since we parse the string above to detect some options

Reply:
the parse only happens for the elif statement above, not inside this elif statement, so method is undefined here. also currently no options affect the sgd method